### ISSUE TITLE
I want to be able to set an aria-label for the checkboxes as its a DAP violation and the sandbox doesnt show / tell me how to do this

### ISSUE THREAD

**Issue by CarolynMabbott** [carbon-design-system/carbon#4461]
I want to be able to set an aria-label for the checkboxes as its a DAP violation and the sandbox doesnt show / tell me how to do this

## Environment

> Operating system
Mac 10.15

> Browser
Google Chrome 

> Automated testing tool and ruleset
DAP

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?
There to be instructions on the site of how to add aria-labels to datatables 

> What happened instead?
There was no instructions to add aria-labels to datatables 

> What WCAG 2.1 checkpoint does the issue violate?
Each form control must have associated label.

## Steps to reproduce the issue

1.  Go to a site with checkboxes
2.  Run DAP
3.  Get error saying its a violation to not have a label
4.  Go to http://react.carbondesignsystem.com/?path=/story/datatable--with-batch-actions to try and find how to add a label
5. There is no instructions to add a label

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

## Additional information

- Screenshots or code
- Notes

**Comment by AlanGreene** [comment]
Can you share the exact violation reported by DAP? The checkbox label does have an aria-label attribute with value 'Select row'

**Comment by CarolynMabbott** [comment]
![Screenshot 2019-10-25 at 15 54 14](https://user-images.githubusercontent.com/32342288/67581497-04ae2500-f740-11e9-9ba3-f5c60aff2c64.png)

It's showing as the area around the checkbox
The exact violation is that `Form control element input is missing associated label.`
